src/core/modules/AtmosphereShowMySky.cpp

```diff
@@ -868,6 +902,7 @@ bool AtmosphereShowMySky::isLoading()
 
 bool AtmosphereShowMySky::isReadyToRender()
 {
+	prevWidth_=0;
```

Comment:
Won't this lead to resize on each frame, even with `ppxatmo==1 && !flagDynamicResolution`?

Reply:
Besides, this function is supposed to be a query (and should have been made `const`, I missed it), so it doesn't seem like a good idea to alter state here.

Reply:
> Won't this lead to resize on each frame, even with `ppxatmo==1 && !flagDynamicResolution`?

it's supposed to trigger a resize when switching from "Preetham" to "ShowMySky".

Reply:
A switch from Preetham to ShowMySky involves construction of a new instance of `AtmosphereShowMySky`, so this assignment should be unnecessary.

Reply:
> A switch from Preetham to ShowMySky involves construction of a new instance of `AtmosphereShowMySky`, so this assignment should be unnecessary.

I just tested it again, without this line I get a wrong display with reduced static resolution.

Reply:
Strange, it behaves identically on my machine. Could you describe your test scenario in detail?

Reply:
- ppxatmo=4
- flag_dynamic_resolution=false

Start Stellarium with "ShowMySky" already activated ~> no problem
Switch to "Preetham" and back to "ShowMySky" ~> Problem ~> Sky graphic is twisted...

Reply:
OK, I can reproduce. In my case I also get segfault because of wrong size of texture obtained in `getMeanPixelValue()`.

What's happening here is that initially, when `AtmosphereShowMySky` is constructed, `resizeRenderTarget()` is called with `ppxatmo=1`, value set from initializer list. Then, when `computeColor()` is called, `ppxatmo` assumes a different value, which is what should trigger a new resize.

Instead, you force the resize on status query, which is not semantically correct.

Reply:
I just moved the reading of the config.ini to the initializer. The problem remains - So I'm back with another suspicious workaround :).

I noticed a second call to "resizeRenderTarget" when starting Stellarium. The first call uses default values for "width" and "height".
When switching from "Preetham" to "ShowMySky" the second call does not happen, the first call already has the actual "width" and "height".

Reply:
OK I found the actual problem, not sure what side to blame. Anyway, `ShowMySky::AtmosphereRenderer` calls `resizeEvent()` using current viewport (`GL_VIEWPORT`) after all data have been loaded (see `stepDataLoading` and its callees).
One fix would be to do the resize after data loading has been done, but it seems kludgy to me. I need some time to check whether this resize at the end is actually needed in the renderer.